test/e2e/lib/servers.ts

```diff
@@ -0,0 +1,146 @@
+import * as http from 'http'
+import { AddressInfo } from 'net'
+import { getIp } from '../../utils'
+import { log } from './logger'
```

Comment:
I have a got a reorganization proposal:

```
- scenario helpers
  - sdkHelpers
  - browserHelpers
  - globalTypes
- testSetup
  - createTest (+ merge declareTest)
  - pageSetups (rename of setups)
  - sdkBuilds
  - servers
    - servers
    - intakeServerApp
    - mockServerApp
  - common
    - logger
    - eventsRegistry
- serverTypes
```
does it make sense?

Reply:
This looks a bit weird to me.

* Why `globalTypes` is with the helpers? They are not helper functions.
* Modules inside `testSetup` are all the code used when setuping a test. But `servers` live accross tests, and the `waitServersIdle` function is used after test setup. I don't think it has its place in `testSetup`.
* The `logger` also have little to do with a single test setup, and it is used by the servers during the execution of all the spec.
* Why the "testSetup/common" folder?

Reply:
nice!

It still a bit hard to me to navigate across this code and I feel like regrouping similar things together could help:

> Why globalTypes is with the helpers? They are not helper functions.

maybe scenario helpers is not the right name, the idea is to regroup utility code for scenarios and isolate it from the "test framework" code 

> Modules inside testSetup are all the code used when setuping a test. But servers live accross tests, and the waitServersIdle function is used after test setup. I don't think it has its place in testSetup.

what about having testSetup (or other name) directory contain all the code related to our "test framework"?
We could then then better identify the main responsibilities:
- test API
- build test page
- build test sdk 
- test servers

> Why the "testSetup/common" folder?

to regroup some less important files used by several others, could be kept at the same level for now

Reply:
I moved things around again, now it looks like this:
```
test/e2e
├── browsers.conf.js
├── lib
│   ├── framework
│   │   ├── createTest.ts
│   │   ├── eventsRegistry.ts
│   │   ├── httpServers.ts
│   │   ├── index.ts
│   │   ├── logger.ts
│   │   ├── pageSetups.ts
│   │   ├── sdkBuilds.ts
│   │   └── serverApps
│   │       ├── intake.ts
│   │       └── mock.ts
│   ├── helpers
│   │   ├── browser.ts
│   │   └── sdk.ts
│   └── types
│       ├── global.ts
│       └── serverEvents.ts
├── scenario
│   ├── internalMonitoring.scenario.ts
│   ├── logs.scenario.ts
│   ├── rum
│   │   ├── actions.scenario.ts
│   │   ├── errors.scenario.ts
│   │   ├── resources.scenario.ts
│   │   └── views.scenario.ts
│   └── tracing.scenario.ts
├── ts-node.js
├── tsconfig.json
├── wdio.base.conf.js
├── wdio.bs.conf.js
└── wdio.local.conf.js

7 directories, 26 files
```
I hope it makes more sense :)